File: Day 1/DFS.py
class Solution:
    def dfs(self,source,adj,ans,visited):
        visited[source]=True
        ans.append(source)
        for neighbour in adj[source]:
            if(visited[neighbour]==False):
                Solution().dfs(neighbour,adj,ans,visited)
    def dfsOfGraph(self, V, adj):
        visited=[False]*V
        ans=list()
        Solution().dfs(0,adj,ans,visited)
        return ans

File: Day 1/DFS.java

class Solution
{
    public ArrayList<Integer> dfsOfGraph(int V, ArrayList<ArrayList<Integer>> adj)
    {
        ArrayList<Integer> ans=new ArrayList<>();
        boolean visited[]=new boolean[V];
        dfs(0,adj,ans,visited);
        return ans;
    }
    public static void dfs(int source,ArrayList<ArrayList<Integer>> adj, ArrayList<Integer> ans,boolean visited[]){
        visited[source]=true;
        ans.add(source);
        for(Integer neighbour:adj.get(source)){
            if(!visited[neighbour])
                dfs(neighbour,adj,ans,visited);
        }
    }
}
